documentation/grammars.md

```diff
@@ -0,0 +1,62 @@
+# Grammar Syntax Documentation
+
+Grammars in Pliant are defined by an bnf dialect similar to abnf and ebnf. Some additional control statements have been added for ignoring whitespace and specifying start productions.
+
+# Grammar of Grammars
+
+The supported language for grammars in Pliant is defined by the following BNF grammar. 
+
+```
+Definition          =   Block
+                        | Block Definition
+
+Block               =   Rule
+                        | Setting
+                        | LexerRule
+
+Rule                =   QualifiedIdentifier '=' Expression ';'
+
+Setting             =   SettingIdentifier '=' QualifiedIdentifier ';'
+
+LexerRule           =   QualifiedIdentifier '~' LexerRuleExpression ';'
+
+Expression          =   Term
```

Comment:
in EBNF it is defined at level of parser, not lexer, so your framework is not powerfull enough to process EBNF